msal/src/main/java/com/microsoft/identity/client/statemachine/states/SignUpStates.kt

```diff
@@ -0,0 +1,632 @@
+                            correlationId = result.correlationId
+                        ),
+                        invalidAttributes = result.invalidAttributes
+                    )
+                }
+                is SignUpCommandResult.AttributesRequired -> {
+                    SignUpResult.AttributesRequired(
+                        nextState = SignUpAttributesRequiredState(
+                            flowToken = result.signupToken,
+                            username = username,
+                            config = config
+                        ),
+                        requiredAttributes = result.requiredAttributes.toListOfRequiredUserAttribute()
+                    )
+                }
+                is SignUpCommandResult.Complete -> {
+                    SignUpResult.Complete(
+                        nextState = SignInAfterSignUpState(
+                            signInVerificationCode = result.signInSLT,
+                            username = username,
+                            config = config
+                        )
+                    )
+                }
+                is INativeAuthCommandResult.Redirect -> {
+                    SignUpResult.BrowserRequired(
+                        error = BrowserRequiredError(
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
+                // This should be caught earlier in the flow, so throwing UnexpectedError
+                is SignUpCommandResult.UsernameAlreadyExists -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result: $result"
+                    )
+                    SignUpResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = result.errorDescription,
+                            error = result.error,
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
+                is INativeAuthCommandResult.UnknownError -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result: $result"
+                    )
+                    SignUpResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = result.errorDescription,
+                            error = result.error,
+                            correlationId = result.correlationId,
+                            details = result.details,
+                            exception = result.exception
+                        )
+                    )
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Native Auth uses a state machine to denote state and transitions for a user.
+ * SignInAfterSignUpState class represents a state where the user must signin after successful
```

Comment:
So we don't automatically sign the user in ourself as part of sign up?

Reply:
From user's perspective we do. The signup operation will provide a short lived token to the sign in process and that will complete the sign in. It will be completely transparent to the user.

Reply:
So the client app developer doesn't have to invoke the SignInAfterSignupState?

Reply:
Yes, the client doesn't have to invoke SignInAfterSignupState

Reply:
I believe we also make sure documentation in customer-facing methods doesn't refer to the SLT token directly, Seems we refere to it as verification code here?

Reply:
I think they would still have to invoke the sign in method from signInAfterSignupState, how else would they pass scopes when signing in.

Reply:
The token names are being changed to continuation_token. I was wrong in the previous comment. The developer has to call signin method after successful signup operation.